src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -738,6 +738,17 @@ static void set_our_sigflags(int sig, int flags) {
   }
 }
 
+// Implementation may use the same storage for both the sa_sigaction field and the sa_handler field,
+// so check for "sigAct.sa_flags == SA_SIGINFO"
+static address get_signal_handler(struct sigaction action) {
```

Comment:
This copies the structure by value. While the compiler will probably optimize this out, I'd still not do it. Please use 
```
static address get_signal_handler(const struct sigaction*) 
``` 
instead.

Reply:
Thank you Thomas for the review.

How about:

`static address get_signal_handler(struct sigaction &action)`

?

Reply:
Meh. I am not a big fan of references :) since at the call site they are indistinguishable from by-value parameters. But if you use them, please make them const.

Reply:
I checked the existing code and decided to use the `get_signal_handler(const struct sigaction* action)` type after all, as that's what all other functions, except for one, use.